src/otr/module.c

```diff
@@ -0,0 +1,269 @@
+{
+	char *new_msg = NULL;
+
+	if (otr_receive(server, msg, nick, &new_msg)) {
+		signal_stop();
+		otrl_message_free(new_msg);
+		return;
+	}
+
+	if (new_msg == NULL) {
+		/* This message was not OTR */
+		signal_continue(4, server, msg, nick, address);
+	} else {
+		/*
+		 * Check for /me IRC marker and if so, handle it so the user does not
+		 * receive a message beginning with /me but rather let irssi handle it
+		 * as a IRC action.
+		 */
+		if (strncmp(new_msg, OTR_IRC_MARKER_ME, OTR_IRC_MARKER_ME_LEN) == 0) {
+			signal_stop();
+			signal_emit("message irc action", 5, server, new_msg + OTR_IRC_MARKER_ME_LEN, nick, address, nick);
+		} else {
+			/* OTR received message */
+			signal_continue(4, server, new_msg, nick, address);
+		}
+	}
+
+	otrl_message_free(new_msg);
+}
+
+/*
+ * Finish an OTR conversation when its query is closed.
+ */
+static void sig_query_destroyed(QUERY_REC *query)
+{
+	if (query && query->server && query->server->connrec) {
+		otr_finish(query->server, query->name);
+	}
+}
+
+/*
+ * Handle /me IRC command.
+ */
+static void cmd_me(const char *data, IRC_SERVER_REC *server,
+		WI_ITEM_REC *item)
+{
+	int ret;
+	const char *target;
+	char *msg, *otrmsg = NULL;
+	QUERY_REC *query;
+
+	query = QUERY(item);
+
+	if (query == NULL || query->server == NULL) {
+		return;
+	}
+
+	CMD_IRC_SERVER(server);
+	if (!IS_IRC_QUERY(query)) {
+		return;
+	}
+
+	if (server == NULL || !server->connected) {
+		cmd_return_error(CMDERR_NOT_CONNECTED);
+	}
+
+	target = window_item_get_target(item);
+
+	ret = asprintf(&msg, OTR_IRC_MARKER_ME "%s", data);
+	if (ret < 0) {
+		return;
+	}
+
+	/* Critical section. On error, message MUST NOT be sent */
+	ret = otr_send(query->server, msg, target, &otrmsg);
+	free(msg);
+
+	if (otrmsg == NULL) {
+		return;
+	}
+
+	signal_stop();
+
+	if (otrmsg) {
+		/* Send encrypted message */
+		irssi_send_message(SERVER(server), target, otrmsg);
+		otrl_message_free(otrmsg);
+	}
+
+	signal_emit("message irc own_action", 3, server, data, item->visible_name);
+}
+
+/*
+ * Optionally finish conversations on /quit. We're already doing this on unload
+ * but the quit handler terminates irc connections before unloading.
+ */
+static void cmd_quit(const char *data, void *server, WI_ITEM_REC *item)
+{
+	otr_finishall(user_state_global);
+}
+
+/*
+ * Create otr module directory if none exists.
+ */
+static void create_module_dir(void)
+{
+	int ret;
+	char *dir_path = NULL;
+	struct stat statbuf;
+
+	/* Create ~/.irssi/otr directory. */
+	ret = asprintf(&dir_path, "%s/%s", get_irssi_dir(), OTR_DIR);
+	if (ret < 0)
+		g_error("Unable to allocate memory for OTR directory path.");
+
+	if (stat(dir_path, &statbuf) != 0) {
+		if (g_mkdir_with_parents(dir_path, 0700) != 0)
+			g_error("Unable to create OTR directory path.");
+	} else if (!S_ISDIR(statbuf.st_mode))
+		g_error("%s is not a directory.\nYou should remove it with command: rm %s", dir_path, dir_path);
```

Comment:
`g_error()` results in `abort()`, hell no

Reply:
Fixed in 7b6cc27d3a447ed9c33fb444df4c7cee6279f59b